crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2918,6 +2918,33 @@ impl Backend {
         ))
     }
 
+    pub fn get_blob_by_tx_hash(&self, hash: B256) -> Result<Option<TxEip4844WithSidecar>> {
+        let tx = self.mined_transaction_by_hash(hash).unwrap();
+        let typed_tx = TypedTransaction::try_from(tx).unwrap();
+        if let Some(sidecar) = typed_tx.sidecar() {
+            return Ok(Some(sidecar.clone()));
+        }
+        Ok(None)
```

Comment:
this can fail, so we should handle this gracefully and only return the blobs Vec<Blob> here